scripts/superdesk/upload/image-crop-directive.js

```diff
@@ -279,136 +278,140 @@
-                        container.prepend(img);
-                        container.append(overlay);
-                        elem.append(container);
-                        drawPoint();
-                    };
-
-                    function updatePOI(offsetX, offsetY) {
-                        scope.$apply(function() {
-                            scope.point.x = Math.round(offsetX * 100 / img.width) / 100;
-                            scope.point.y = Math.round(offsetY * 100 / img.height) / 100;
-                            scope.onChange();
-                        });
+            }],
+            link: function(scope, element, attrs, vm) {
+                var circleRadius = 30 / 2;
+                var lineThickness = 2;
+                var $poiContainer = element.find('.image-point__poi');
+                var $poiCursor = element.find('.image-point__poi__cursor');
+                var $poiLeft = element.find('.image-point__poi__cross-left');
+                var $poiRight = element.find('.image-point__poi__cross-right');
+                var $poiTop = element.find('.image-point__poi__cross-top');
+                var $poiBottom = element.find('.image-point__poi__cross-bottom');
+                function drawPoint(img, poi) {
+                    if (!angular.isDefined(poi)) {
+                        poi = vm.poi;
                     }
-
-                    function leaveOverlay(e) {
-                        e.preventDefault();
-                        updatePOI(e.offsetX, e.offsetY);
-                        $rootScope.$broadcast('poiUpdate', scope.point);
-                        overlay.onmousemove = null;
-                        overlay.onmouseup = null;
-                        overlay.onmouseleave = null;
-                    }
-
-                    overlay.onmousedown = function(e) {
-                        e.preventDefault();
-                        updatePOI(e.offsetX, e.offsetY);
-                        $rootScope.$broadcast('poiUpdate', scope.point);
-
-                        overlay.onmousemove = function(e) {
-                            e.preventDefault();
-                            updatePOI(e.offsetX, e.offsetY);
-                        };
-
-                        overlay.onmouseup = leaveOverlay;
-                        overlay.onmouseleave = leaveOverlay;
-                    };
-
-                    img.src = src;
-
-                    container.append(pointElem);
-                    container.append(crossLeft);
-                    container.append(crossRight);
-                    container.append(crossTop);
-                    container.append(crossBottom);
-
-                }
-
-                function drawPoint() {
-                    pointElem.css({
-                        left: (scope.point.x * img.width) - 15,
-                        top: (scope.point.y * img.height) - 15
+                    var topOffset = (poi.y * img.height) - circleRadius;
```

Comment:
These vars can also be declared at the top of the link function to avoid redeclaring a var each time which is much more expensive.